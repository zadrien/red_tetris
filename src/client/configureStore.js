import { createBrowserHistory } from 'history'
import { combineReducers, applyMiddleware, compose, createStore } from 'redux'
import { connectRouter, routerMiddleware } from 'connected-react-router'

import thunk from 'redux-thunk'
import { createLogger } from 'redux-logger'

import socketMiddleware from './middleware/socketMiddleware'

import TetrisApp from './reducers'

const createRootReducer = (history) => combineReducers({
	router: connectRouter(history),
	TetrisApp
})

export const history = createBrowserHistory()

export default function configureStore(preloadedState) {
	const store = createStore(
	  createRootReducer(history),
	  preloadedState,
	  applyMiddleware(
			routerMiddleware(history),
			thunk,
			createLogger(),
			socketMiddleware()
		),
	)

	return store
}
