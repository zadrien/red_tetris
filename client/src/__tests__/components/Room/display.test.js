import React from 'react'

import { Provider } from 'react-redux'
import renderer from 'react-test-renderer'
import configureStore from 'redux-mock-store'

import Display from '../../../components/Room/MainDisplay'

const mockStore = configureStore([])

describe('Main User Display TDD', () => {
	let store, component
	const renderComponent = (store) => {
		return renderer.create(
			<Provider store={store}>
				<Display/>
			</Provider>
		)
	}

	const getStore = (state) => mockStore(state)

	it('should render (display as undefined)', () => {
		const state = {
			room: {
				  start: false,
				  host: true,
			}
		}
		
		component = renderComponent(getStore(state))
		const testInstance = component.root

		expect(component.toJSON()).toMatchSnapshot()
		expect(testInstance.findByProps({className: 'board'}).children.length).toEqual(0)
        
	})

    it('should render (start as true)', () => {
		const state = {
			room: {
				display: [
					[ '.', '.', '.', '.', '.', '.', '.', '.', '.', '.' ],
					[ '.', '.', '.', '.', '.', '.', '.', '.', '.', '.' ],
					[ '.', '.', '.', '.', 'X', '.', '.', '.', '.', '.' ],
					[ '.', '.', '.', '.', '.', '.', '.', '.', '.', '.' ],
					[ '.', '.', '.', '.', '.', '.', '.', '.', '.', '.' ]
				  ],
				  start: true,
				  host: true,
			}
		}
		
		component = renderComponent(getStore(state))
		const testInstance = component.root

		expect(component.toJSON()).toMatchSnapshot()
		expect(testInstance.findByProps({className: 'board'}).children.length).toEqual(5)
        
	})
	
	it('should render (start and host as false)', () => {
		const state = {
			room: {
				display: [
					[ '.', '.', '.', '.', '.', '.', '.', '.', '.', '.' ],
					[ '.', '.', '.', '.', '.', '.', '.', '.', '.', '.' ],
					[ '.', '.', '.', '.', 'X', '.', '.', '.', '.', '.' ],
					[ '.', '.', '.', '.', '.', '.', '.', '.', '.', '.' ],
					[ '.', '.', '.', '.', '.', '.', '.', '.', '.', '.' ]
				  ],
				  start: false,
				  host: false,
			}
		}
		
		component = renderComponent(getStore(state))
		const testInstance = component.root

		expect(component.toJSON()).toMatchSnapshot()
		expect(testInstance.findByProps({className: 'board'}).children.length).toEqual(6)
		expect(testInstance.findByProps({className: "title medium"}).children).toEqual(["Pending..."])
	})
	
	it('should render (winner as true)', () => {
		const state = {
			room: {
				display: [
					[ '.', '.', '.', '.', '.', '.', '.', '.', '.', '.' ],
					[ '.', '.', '.', '.', '.', '.', '.', '.', '.', '.' ],
					[ '.', '.', '.', '.', 'X', '.', '.', '.', '.', '.' ],
					[ '.', '.', '.', '.', '.', '.', '.', '.', '.', '.' ],
					[ '.', '.', '.', '.', '.', '.', '.', '.', '.', '.' ]
				  ],
				  start: false,
				  host: false,
				  winner: true
			}
		}
		
		component = renderComponent(getStore(state))
		const testInstance = component.root

		expect(component.toJSON()).toMatchSnapshot()
		expect(testInstance.findByProps({className: 'board'}).children.length).toEqual(6)
		expect(testInstance.findByProps({className: "title big"}).children).toEqual(["GAME OVER"])
		expect(testInstance.findByProps({className: "title medium"}).children).toEqual(["You won"])
    })    
})