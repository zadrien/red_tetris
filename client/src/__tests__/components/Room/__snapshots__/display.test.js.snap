// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main User Display TDD should render (display as undefined) 1`] = `
<div
  className="board"
/>
`;

exports[`Main User Display TDD should render (start and host as false) 1`] = `
<div
  className="board"
>
  <div
    className="board-overlay transparent d-flex col space-around"
  >
    <h6
      className="title medium"
    >
      Pending...
    </h6>
  </div>
  <div
    className="board-row"
  >
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
  </div>
  <div
    className="board-row"
  >
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
  </div>
  <div
    className="board-row"
  >
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": undefined,
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
  </div>
  <div
    className="board-row"
  >
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
  </div>
  <div
    className="board-row"
  >
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
  </div>
</div>
`;

exports[`Main User Display TDD should render (start as true) 1`] = `
<div
  className="board"
>
  <div
    className="board-row"
  >
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
  </div>
  <div
    className="board-row"
  >
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
  </div>
  <div
    className="board-row"
  >
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": undefined,
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
  </div>
  <div
    className="board-row"
  >
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
  </div>
  <div
    className="board-row"
  >
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
  </div>
</div>
`;

exports[`Main User Display TDD should render (winner as true) 1`] = `
<div
  className="board"
>
  <div
    className="board-overlay transparent d-flex col space-around"
  >
    <h6
      className="title big"
    >
      GAME OVER
    </h6>
    <h6
      className="title medium"
    >
      You won
    </h6>
    <button
      className="bob-btn"
      onClick={[Function]}
    >
      OK
    </button>
  </div>
  <div
    className="board-row"
  >
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
  </div>
  <div
    className="board-row"
  >
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
  </div>
  <div
    className="board-row"
  >
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": undefined,
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
  </div>
  <div
    className="board-row"
  >
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
  </div>
  <div
    className="board-row"
  >
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
    <div
      className="board-box"
      style={
        Object {
          "backgroundColor": "Black",
        }
      }
    />
  </div>
</div>
`;
